src/main/java/com/dynatrace/openkit/protocol/Beacon.java

```diff
@@ -291,15 +295,21 @@ public void endSession(SessionImpl session) {
             return;
         }
 
+        if(beaconConfiguration.get().getDataCollectionLevel() == DataCollectionLevel.OFF) {
+            return;
+        }
+
         StringBuilder eventBuilder = new StringBuilder();
 
         buildBasicEventData(eventBuilder, EventType.SESSION_END, null);
 
         addKeyValuePair(eventBuilder, BEACON_KEY_PARENT_ACTION_ID, 0);
         addKeyValuePair(eventBuilder, BEACON_KEY_START_SEQUENCE_NUMBER, createSequenceNumber());
-        addKeyValuePair(eventBuilder, BEACON_KEY_TIME_0, getTimeSinceSessionStartTime(session.getEndTime()));
 
-        addEventData(session.getEndTime(), eventBuilder);
+        long sessionEndTime = session.getEndTime();
```

Comment:
please revert this change and only use getEndTime here. the local var makes it harder to read